controllers/rack.go

```diff
@@ -420,7 +424,15 @@ func (r *SingleClusterReconciler) upgradeOrRollingRestartRack(found *appsv1.Stat
 	}
 
 	if r.aeroCluster.Spec.RackConfig.MaxIgnorablePods != nil {
-		if res := r.handleNSOrDeviceRemovalForIgnorablePods(rackState, ignorablePodNames); !res.isSuccess {
+		if res = r.handleNSOrDeviceRemovalForIgnorablePods(rackState, ignorablePodNames); !res.isSuccess {
+			return found, res
+		}
+	}
+
+	// handle k8sNodeBlockList pods only if it is changed
+	if !reflect.DeepEqual(r.aeroCluster.Spec.K8sNodeBlockList, r.aeroCluster.Status.K8sNodeBlockList) {
```

Comment:
If this field is added along with some operation like upgrade or rolling restart then flow will go inside the function but there won't be any pod to migrate, correct? We should cover this in our tests, if not covered already.